app/controllers/api/v2/anime_controller.rb

```diff
@@ -2,6 +2,21 @@ module Api::V2
   class AnimeController < ApiController
     caches_action :show, expires_in: 1.hour
 
+    def index
+      anime = Anime.preload(:genres)
+
+      if params.key? :filter
+        search = AnimeSearch.new(params[:filter])
+        if search.valid?
+          anime = search.apply_filters(anime)
+        else
+          return render json: search.errors.full_messages
+        end
+      end
+
+      render json: anime.page(params[:page])
```

Comment:
It doesn't throw any errors without `order`, but the behavior is undefined — database offsets don't make sense without an order and can potentially return random results.

As for the `per`, we should be able to use a `default_scope`